### PR TITLE
MGMT-10817: Fix install-config on vsphere platform

### PR DESCRIPTION
**Pull request by eliorerz**
- Separate none platform install-config from baremetal.
- Add support for platforms alongside enabled user-managed-networking

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @tsorya 
/cc @gamli75 
/cc @vrutkovs

### CODE REVIEW

**Comment by eliorerz** [comment]
> /test edge-e2e-vsphere-assisted-umn

It will not work.. we still need to create and configure the dns server on test-infra, do a few fixes on the vsphere terraform and more

File: internal/provider/vsphere/installConfig.go [hunk 1]
```diff
@@ -18,18 +19,26 @@ func setPlatformValues(platform *installcfg.VsphereInstallConfigPlatform) {
...
-		return errors.New("invalid cluster parameters, IngressVip must be provided")
-	}
-	vsPlatform := &installcfg.VsphereInstallConfigPlatform{
-		APIVIP:     cluster.APIVip,
-		IngressVIP: cluster.IngressVip,
+
+	vsPlatform := &installcfg.VsphereInstallConfigPlatform{}
+
+	if !swag.BoolValue(cluster.UserManagedNetworking) {
+		if len(cluster.APIVip) == 0 {
```
Comment (eranco74) [inline internal/provider/vsphere/installConfig.go:30]: Didn't you validate the same condition in line 23?

Comment (eranco74) [reply internal/provider/vsphere/installConfig.go:30]: I guess you want to remove line 23, this seems better (handles ingress as well)

Comment (eliorerz) [reply internal/provider/vsphere/installConfig.go:30]: Yeah, thanks..

**Comment by tsorya** [comment]
still need unitests right?

File: internal/provider/vsphere/installConfig.go [hunk 2]
```diff
@@ -18,18 +19,26 @@ func setPlatformValues(platform *installcfg.VsphereInstallConfigPlatform) {
...
-	if len(cluster.IngressVip) == 0 {
-		return errors.New("invalid cluster parameters, IngressVip must be provided")
-	}
-	vsPlatform := &installcfg.VsphereInstallConfigPlatform{
-		APIVIP:     cluster.APIVip,
-		IngressVIP: cluster.IngressVip,
+
+	vsPlatform := &installcfg.VsphereInstallConfigPlatform{}
+
+	if !swag.BoolValue(cluster.UserManagedNetworking) {
```
Comment (eranco74) [inline internal/provider/vsphere/installConfig.go:29]: Please add a UT for this case

**Comment by eliorerz** [comment]
> still need unitests right?

Yes.. this is why the PR is still in WIP

File: internal/provider/vsphere/installConfig.go (hunk 2 above)
Comment (eliorerz) [reply internal/provider/vsphere/installConfig.go:25]: https://github.com/openshift/assisted-service/pull/4384#issuecomment-1243648980

Comment (eliorerz) [reply internal/provider/vsphere/installConfig.go:25]: But we do have UT for this case .. every install-config none-platform test is checking that

**Comment by eliorerz** [comment]
None platform [install-config](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_assisted-service/4384/pull-ci-openshift-assisted-service-master-edge-e2e-metal-assisted-none/1569318873470078976/artifacts/e2e-metal-assisted-none/assisted-common-gather/artifacts/2022-09-12_14:27:33_8bf8363e-6951-4266-9edf-673d5012cda6/cluster_files/install-config.yaml) is OK

**Comment by eliorerz** [comment]
Had to rebase 

/test edge-e2e-metal-assisted-none edge-e2e-vsphere-assisted edge-e2e-metal-assisted-single-node

**Comment by eliorerz** [comment]
`unittests` and `lint` pass locally. `e2e-metal-assisted-none` failed to setup packet machine. Retesting..
/retest

**Comment by eliorerz** [comment]
/test edge-e2e-vsphere-assisted-umn
We might get the cluster logs/install-config

**Comment by eliorerz** [comment]
Install config was verified locally (deployed assisted service) for vsphere + UMN enabled:
```
"apiVersion: v1\nbaseDomain: redhat.com\nnetworking:\n  networkType: OpenShiftSDN\n  clusterNetwork: []\n  serviceNetwork: []\nmetadata:\n  name: test-infra-cluster-3a0941fd\ncompute:\n- hyperthreading: Enabled\n  name: worker\n  replicas: 0\ncontrolPlane:\n  hyperthreading: Enabled\n  name: master\n  replicas: 0\nplatform:\n  vsphere:\n    vCenter: vcenterplaceholder
```

**Pull request merged**